lib/faster_s3_url/builder.rb

```diff
@@ -103,34 +104,24 @@ def presigned_url(key, time: nil, expires_in: DEFAULT_EXPIRES_IN,
 
       credential_scope = datestamp + '/' + region + '/' + SERVICE + '/' + 'aws4_request'
 
-      canonical_query_string_parts = [
-          "X-Amz-Algorithm=#{ALGORITHM}",
-          "X-Amz-Credential=" + uri_escape(@access_key_id + "/" + credential_scope),
-          "X-Amz-Date=" + amz_date,
-          "X-Amz-Expires=" + expires_in.to_s,
-          "X-Amz-SignedHeaders=" + SIGNED_HEADERS,
-        ]
-
-      extra_params = {
-        :"response-cache-control" => response_cache_control,
-        :"response-content-disposition" => response_content_disposition,
-        :"response-content-encoding" => response_content_encoding,
-        :"response-content-language" => response_content_language,
-        :"response-content-type" => response_content_type,
-        :"response-expires" => convert_for_timestamp_shape(response_expires),
-        :"versionId" => version_id
+      # These have to be sorted, but sort is case-sensitive, and we have a fixed
+      # list of headers we know might be here... turns out they are already sorted?
+      canonical_query_params = {
+        'X-Amz-Algorithm': ALGORITHM,
```

Comment:
I noticed that I didn't consistently use double quotes here. It seems like the pattern of the file is to use double quotes for everything. I'll change this line and update my commit.

Reply:
honestly on small projects like this I don't really worry about that